modules/xmpp/JingleSessionPC.js

```diff
@@ -503,25 +503,48 @@ export default class JingleSessionPC extends JingleSession {
             // IS_SELECTED_CHANGED events and notify the peer connection
             this.rtc.addListener(RTCEvents.IS_SELECTED_CHANGED,
                 isSelected => {
-                    this.peerconnection.setIsSelected(isSelected);
-                    logger.info('Doing local O/A due to '
-                        + 'IS_SELECTED_CHANGED event');
-                    this.modificationQueue.push(finishedCallback => {
-                        if (!this._assertNotEnded('setIsSelected O/A')) {
-                            finishedCallback();
-
-                            return;
-                        }
+                    this._executeOnTheQueue('setIsSelected', finishedCallback => {
+                        logger.info('Doing local O/A due to IS_SELECTED_CHANGED event');
+                        this.peerconnection.setIsSelected(isSelected);
 
                         this._renegotiate()
-                            .then(finishedCallback)
-                            .catch(finishedCallback);
+                            .then(finishedCallback, finishedCallback);
```

Comment:
personally i prefer the old syntax (explicitly setting `then` and `catch`) because it's clearer but maybe that's just me.

Reply:
So `then` and `catch` are technically two different things, doing `.catch` on `.then` will also catch any rejections that would happen in the `.then` chain. And usually it is not the intention of the person who writes it from my experience.

Reply:
It is obviously not the case in a scenario simple like this one

Reply:
if someone wrote catch, i assume they mean to catch.  here, if an exception is throw in the then it'll be unhandled which doesn't seem good?

Reply:
actually i guess we'd throw twice here (and execute the same method twice), so ignore what i said